sourcecode/MSG_GAN/Losses.py

```diff
@@ -97,15 +97,22 @@ def __gradient_penalty(self, real_samps, fake_samps, reg_lambda=10):
         :return: tensor (gradient penalty)
         """
 
-        batch_size = real_samps.shape[0]
+        def _interp(real_samp, fake_samp):
+            batch_size = real_samp.shape[0]
 
-        # generate random epsilon
-        epsilon = th.rand((batch_size, 1, 1, 1)).to(fake_samps.device)
+            # generate random epsilon
+            epsilon = th.rand((batch_size, 1, 1, 1)).to(fake_samp.device)
```

Comment:
There is a tine issue here. The epsilon would be different for different `real_samp` and `fake_samp`, which is not intended; and perhaps also incorrect. Could you add the epsilon as another parameter to this `_interp` function and then select an epsilon only once outside? Basically the epsilon should be same for interpolating real and fake samples at all scales. 
Rest everything looks good. :+1:. Thanks.

Reply:
Ok， I will try it

Reply:
@Johnson-yue @akanimax any result of this? This PR is still open?

Reply:
Hi， I try to adding gp in MSG-GAN, but, it is confusion, the  multi-output need gradient for D ,so it can be compute

Reply:
@cellis